embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/testcases/SessionApiTest.kt

```diff
@@ -50,20 +50,27 @@ internal class SessionApiTest {
 
             // validate snapshots separately, as the JSON diff is tricky to debug
             val snapshots = checkNotNull(message.data.spanSnapshots)
-            assertEquals(2, snapshots.size)
+            assertEquals(1, snapshots.size)
 
             // validate network status span
             val networkStatusSpan = snapshots.single { it.name == "emb-network-status" }
             assertEquals(startTime, networkStatusSpan.startTimeNanos?.nanosToMillis())
             assertEquals("sys.network_status", networkStatusSpan.attributes?.findAttributeValue("emb.type"))
 
             // validate session span
-            val sessionSpan = snapshots.single { it.name == "emb-session" }
+            val spans = checkNotNull(message.data.spans)
```

Comment:
We were validating the session based on the snapshot. No wonder it was flakey before - it wasn't even the final state of session, just what the last cached version was